app/src/main/java/org/mozilla/FirefoxLite/firefoxdownloader/SampleService.java

```diff
@@ -0,0 +1,249 @@
+package org.mozilla.FirefoxLite.firefoxdownloader;
+
+import android.app.Notification;
+import android.app.Service;
+import android.content.Intent;
+import android.content.IntentFilter;
+import android.content.SharedPreferences;
+import android.graphics.Color;
+import android.os.IBinder;
+import android.preference.PreferenceManager;
+import android.util.JsonReader;
+import android.util.Log;
+import android.widget.Toast;
+import androidx.annotation.Nullable;
+import androidx.core.app.NotificationCompat;
+import androidx.lifecycle.LifecycleOwner;
+import androidx.lifecycle.LifecycleService;
+
+import static com.sample.NotificationHelper.channel1ID;
+
+public class SampleService extends LifecycleService {
+    final static String MY_ACTION = "MY_ACTION";
+
+    final String URL1 = "http://mattmahoney.net/dc/enwik8.pmd";
+    private static String dirPath;
+    int downloadIdOne;
+    NotificationHelper mNotificationHelper;
+    MainActivity mainActivity;
```

Comment:
may I know what is this for?

Reply:
I will remove the main activity object. Notification helper is for notification.DownloadIdOne is the  id of first download.Dirpath is path where download is stored